src/avro/src/schema.rs

```diff
@@ -262,6 +271,23 @@ pub enum SchemaPiece {
     },
 }
 
+impl SchemaPiece {
+    /// Returns whether the schema node is "underlyingly" an Int (but possibly a logicalType typedef)
```

Comment:
perhaps `is internally` is better than `is underlyingly` but then again shakespeare made up words too :D no actual preference do what feels best to you here